dbt/models/qc/qc.vw_report_town_close_res_edit.sql

```diff
@@ -0,0 +1,193 @@
+-- Get up to the first five DWELDAT cards for each parcel
+WITH first_five_dwellings AS (
+    SELECT
+        parid,
+        taxyr,
+        card,
+        yrblt,
+        sfla,
+        stories,
+        rmbed,
+        fixbath,
+        fixhalf,
+        bsmt,
+        user12 AS bsmt_fin,
+        external_propct,
+        user16 AS alt_cdu,
+        row_num
+    FROM (
+        SELECT
+            *,
+            ROW_NUMBER()
+                OVER (PARTITION BY parid, taxyr ORDER BY card ASC)
+                AS row_num
+        FROM {{ source('iasworld', 'dweldat') }}
+        WHERE cur = 'Y'
+            AND deactivat IS NULL
+    ) AS ranked_dwellings
+    WHERE row_num <= 5
+),
+
+-- Pivot the dwelling chars using the MAP_AGG function and key access
+-- as recommended here:
+-- https://gist.github.com/shotahorii/6b710c902a8a6ef184987ca787d329d9
+first_five_dwellings_pivoted AS (
+    SELECT
+        parid,
+        taxyr,
+    {% for idx in range(1, 6) %}
+        yrblt_kv[{{ idx }}] AS yrblt_{{ idx }},
+        sfla_kv[{{ idx }}] AS sfla_{{ idx }},
+        stories_kv[{{ idx }}] AS stories_{{ idx }},
+        rmbed_kv[{{ idx }}] AS rmbed_{{ idx }},
+        fixbath_kv[{{ idx }}] AS fixbath_{{ idx }},
+        fixhalf_kv[{{ idx }}] AS fixhalf_{{ idx }},
+        bsmt_kv[{{ idx }}] AS bsmt_{{ idx }},
+        bsmt_fin_kv[{{ idx }}] AS bsmt_fin_{{ idx }},
+        external_propct_kv[{{ idx }}] AS external_propct_{{ idx }},
+        alt_cdu_kv[{{ idx }}] AS alt_cdu_{{ idx }}{% if not loop.last %}
+            ,
+        {% endif %}
+    {% endfor %}
+    FROM (
+        SELECT
+            parid,
+            taxyr,
+            MAP_AGG(row_num, yrblt) AS yrblt_kv,
+            MAP_AGG(row_num, sfla) AS sfla_kv,
+            MAP_AGG(row_num, stories) AS stories_kv,
+            MAP_AGG(row_num, rmbed) AS rmbed_kv,
+            MAP_AGG(row_num, fixbath) AS fixbath_kv,
+            MAP_AGG(row_num, fixhalf) AS fixhalf_kv,
+            MAP_AGG(row_num, bsmt) AS bsmt_kv,
+            MAP_AGG(row_num, bsmt_fin) AS bsmt_fin_kv,
+            MAP_AGG(row_num, external_propct) AS external_propct_kv,
+            MAP_AGG(row_num, alt_cdu) AS alt_cdu_kv
+        FROM first_five_dwellings
+        GROUP BY parid, taxyr
+    ) AS agg_dwellings
+),
+
+num_dwellings AS (
+    SELECT
+        parid,
+        taxyr,
+        COUNT(*) AS num_dwellings
+    FROM first_five_dwellings
+    GROUP BY parid, taxyr
+)
```

Comment:
Note that this is counting the number of dwellings _in the first five dwellings_, and not necessarily the number of dwellings in the parcel. I think this is probably fine, but it's a distinction to keep in mind, and it may make sense for me to double check with Will.

Reply:
Makes sense, I switched to counting all cards in `dweldat` in 83ddeb8fdf2a5472f49bb0f152f56ddf0d58d61b and spot-checked the counts between the raw query and the view output.